### PR TITLE
CSHARP-4851: Render full form for $and/$eq in $vectorSearch stage filter

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/FilterDefinitionRenderContextTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    public class FilterDefinitionRenderContextTests
+    {
+        [Fact]
+        public void FilterDefinitionRenderContext_fullForm_default_value_should_be_false()
+        {
+            FilterDefinitionRenderContext.RenderFullForm.Should().BeFalse();
+        }
+
+        [Fact]
+        public async Task FilterDefinitionRenderContext_should_be_scoped_to_task()
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/FilterDefinitionRenderContextTests.cs:35]: LMK if you think this test is an overkill.

File: src/MongoDB.Driver/FilterDefinition.cs [hunk 2]
```diff
@@ -285,4 +286,30 @@ public override BsonDocument Render(IBsonSerializer<TDocument> documentSerialize
             return new BsonDocumentWrapper(_obj, serializer);
         }
     }
+
+    internal static class FilterDefinitionRenderContext
+    {
+        private static readonly AsyncLocal<bool> __renderFullForm = new AsyncLocal<bool>();
+
+        public static bool RenderFullForm
```
[the remark this answers is not in the document]
Comment (BorisDog) [reply src/MongoDB.Driver/FilterDefinition.cs]: Done.

File: tests/MongoDB.Driver.Tests/FilterDefinitionRenderContextTests.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            var documentSerializer = serializerRegistry.GetSerializer<BsonDocument>();
+            var expectedFilterDocument = BsonDocument.Parse(expectedFilter);
+
+            using var renderContext = FilterDefinitionRenderContext.StartRender(renderFullForm);
+            var actualFilter = filterDefinition.Render(documentSerializer, serializerRegistry);
+
+            actualFilter.Should().Be(expectedFilterDocument);
+        }
+
+        public static IEnumerable<object[]> Correct_form_should_be_rendered_test_cases()
```
[the remark this answers is not in the document]
Comment (BorisDog) [reply tests/MongoDB.Driver.Tests/FilterDefinitionRenderContextTests.cs:80]: Good idea, thanks, done.

**Review by sanych-sun — APPROVED** [review]
LGTM + please make sure we have a ticket to remove this workaround in future version of the driver.

**Pull request merged**